main.go

```diff
@@ -38,58 +48,149 @@ var (
+	}
+	porcelainFlag := cli.BoolFlag{
+		Name:        "porcelain",
+		Usage:       "Print the output of the command in a easy-to-parse format for scripts",
+		Required:    false,
+		Destination: &PorcelainMode,
+		EnvVar:      DefaultPorcelainEnvironmentVariable,
+	}
+	cacheFlag := cli.BoolFlag{
+		Name:        "useCache",
+		Usage:       "Experimental: use a cache to speed up scripts",
+		Required:    false,
+		Destination: &CacheMode,
+		EnvVar:      DefaultCacheEnvironmentVariable,
+	}
 	app.Commands = []cli.Command{
+		{
+			Name:   "list",
+			Usage:  "Print a list of all packages tracked in the current repository",
+			Action: listPackages,
+			Flags:  []cli.Flag{packageFlag, porcelainFlag},
+		},
 		{
 			Name:   "prepare",
 			Usage:  "Pull in the chart specified from upstream to the charts directory and apply any patch files",
 			Action: prepareCharts,
-			Flags:  []cli.Flag{packageFlag},
+			Before: setupCache,
+			Flags:  []cli.Flag{packageFlag, cacheFlag},
 		},
 		{
 			Name:   "patch",
 			Usage:  "Apply a patch between the upstream chart and the current state of the chart in the charts directory",
 			Action: generatePatch,
-			Flags:  []cli.Flag{packageFlag},
+			Before: setupCache,
+			Flags:  []cli.Flag{packageFlag, cacheFlag},
 		},
 		{
 			Name:   "charts",
 			Usage:  "Create a local chart archive of your finalized chart for testing",
 			Action: generateCharts,
-			Flags:  []cli.Flag{packageFlag, configFlag},
+			Before: setupCache,
+			Flags:  []cli.Flag{packageFlag, configFlag, cacheFlag},
+		},
+		{
+			Name:   "index",
+			Usage:  "Creates or updates existing Helm index.yaml at repository root",
+			Action: createOrUpdateIndex,
+		},
+		{
+			Name:   "zip",
+			Usage:  "Takes the contents of a chart under charts/ and rezips the asset if it has changed",
+			Action: zipCharts,
+			Flags:  []cli.Flag{chartFlag},
+		},
+		{
+			Name:   "unzip",
+			Usage:  "Takes the contents of an asset under assets/ and unzips the chart.",
+			Action: unzipAssets,
+			Flags:  []cli.Flag{assetFlag},
 		},
 		{
 			Name:   "clean",
 			Usage:  "Clean up your current repository to get it ready for a PR",
-			Action: cleanRepository,
+			Action: cleanRepo,
 			Flags:  []cli.Flag{packageFlag},
 		},
+		{
+			Name:   "clean-cache",
+			Usage:  "Experimental: Clean cache",
+			Action: cleanCache,
+		},
 		{
 			Name:   "validate",
 			Usage:  "Run validation to ensure that contents of assets and charts won't overwrite released charts",
 			Action: validateRepo,
+			Flags: []cli.Flag{packageFlag, configFlag, cli.BoolFlag{
+				Name:        "local,l",
+				Usage:       "Only perform local validation of the contents of assets and charts",
+				Required:    false,
+				Destination: &LocalMode,
+			}, cli.BoolFlag{
+				Name:        "remote,r",
+				Usage:       "Only perform upstream validation of the contents of assets and charts",
+				Required:    false,
+				Destination: &RemoteMode,
+			},
+			},
```

Comment:
I thought about adding environment variables for this to allow us to pass it in via `make`, but I decided against it since 1) I didn't want to add more env vars and 2) it's a highly specialized command only meant to be used for debugging what is going on with local / remote validation. In general, the average developer doesn't ever have to touch this.

So if you wanted to set it, you'd avoid using `make` and instead just call `./bin/charts-build-scripts validate --local` or `./bin/charts-build-scripts validate --remote`.

Reply:
> So if you wanted to set it, you'd avoid using make and instead just call ./bin/charts-build-scripts validate --local or ./bin/charts-build-scripts validate --remote.

Can we add it to the experimental.yaml?

Reply:
yeah sure